controllers/vpcendpoint/validation.go

```diff
@@ -108,6 +123,16 @@ func (r *VpcEndpointReconciler) validateAWSResources(
 	validationFuncs []ValidateAWSResourceFunc) error {
 	for _, validationFunc := range validationFuncs {
 		if err := validationFunc(ctx, resource); err != nil {
+			if err := r.Status().Update(ctx, resource); err != nil {
+				r.log.V(0).Error(err, "failed to update status")
+				return err
+			}
+
+			return err
```

Comment:
Does this return the error state of validationFunc(ctx, resource) or r.Status().Update(ctx, resource)?

Reply:
`validationFunc()`, e.g. https://go.dev/play/p/67w2iSgg3KX